bottlerocket/types/src/agent_config.rs

```diff
@@ -65,6 +64,63 @@ pub struct SonobuoyConfig {
 
 impl Configuration for SonobuoyConfig {}
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub enum SonobuoyPluginConfig {
+    /// The path to a sonobuoy plugin i.e (https://raw.githubusercontent.com/vmware-tanzu/sonobuoy-plugins/master/cis-benchmarks/kube-bench-plugin.yaml)
+    Path(String),
+    /// Base64 encoded sonobuoy custom plugin yaml.
+    EncodedYaml { name: String, encoded_yaml: String },
+    /// A simple custom sonobuoy plugin.
+    CustomPlugin {
+        name: String,
+        image: String,
+        result_format: ResultFormat,
+    },
+    /// e2e sonobuoy testing
+    E2E(SonobuoyMode),
```

Comment:
nit: I think technically per Rust guidelines the last e should be lowercase.

```suggestion
    E2e(SonobuoyMode),
```